### PR TITLE
[request] Move configuration to PHP

### PR DESCRIPTION
**Pull request by dangkhoagms**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Part of #37186
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -282,7 +282,7 @@ public function load(array $configs, ContainerBuilder $container)
         }
 
         if ($this->isConfigEnabled($container, $config['request'])) {
-            $this->registerRequestConfiguration($config['request'], $container, $loader);
+            $this->registerRequestConfiguration($config['request'], $container, $phpLoader);
```
Comment (rimoi) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:285]: Your variable `$phpLoader`,  it's not defined

Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:285]: it is defined on top

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 2]
```diff
@@ -980,10 +980,10 @@ private function registerSessionConfiguration(array $config, ContainerBuilder $c
         $container->setParameter('session.metadata.update_threshold', $config['metadata_update_threshold']);
     }
 
-    private function registerRequestConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
+    private function registerRequestConfiguration(array $config, ContainerBuilder $container, PhpFileLoader $phpLoader)
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:983]: please don't change the variable name

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+use function Symfony\Component\DependencyInjection\Loader\Configurator\abstract_arg;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+use Symfony\Component\HttpKernel\EventListener\AddRequestFormatsListener;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('request.add_request_formats_listener')
+            ->class(AddRequestFormatsListener::class)
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php:19]: please set the class as second argument to `set`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            ->tag('kernel.event_subscriber')
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php:20]: `tag` after `args` pls

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            ->args([abstract_arg('event subscriber')]);
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php:21]: `abstract_arg('formats')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use function Symfony\Component\DependencyInjection\Loader\Configurator\abstract_arg;
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php:12]: the file should be in the Symfony\Component\DependencyInjection\Loader\Configurator namespace. No need to import the function then.

**Review by Tobion — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php (hunk 5 above)
Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php:21]: the semicolon should be on the next line

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 1 above)
Comment (dangkhoagms) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:285]: > Your variable `$phpLoader`, it's not defined

it is defined in line 174

**Review by Tobion — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Component\HttpKernel\EventListener\AddRequestFormatsListener;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('request.add_request_formats_listener', AddRequestFormatsListener::class)
+            ->args([abstract_arg('formats')])
+            ->tag('kernel.event_subscriber');
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/request.php:20]: the simicolon should be on the next line

**Review by Tobion — CHANGES_REQUESTED** [review]

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @dangkhoagms.

**Pull request merged**